File: Data Structures & Algorithms/jump-game/submission-1.java
class Solution {
    public boolean canJump(int[] nums) {
        int currMax = nums[0], curr = 0;

        while (currMax < nums.length && curr <= currMax) {
            currMax = Math.max(currMax, nums[curr] + curr);
            curr++;
        }

        return currMax >= nums.length - 1;
    }
}

File: Data Structures & Algorithms/subsets/submission-1.go
func subsets(nums []int) [][]int {
    n, sol, curr := len(nums), [][]int{}, []int{}

    var bt func (i int)
    bt = func (i int) {
        if i == n {
            sol = append(sol, append([]int{}, curr...))
            return
        }
        curr = append(curr, nums[i])
        bt(i+1)
        curr = curr[:len(curr)-1]
        bt(i+1)
    }

    bt(0)
    return sol
}
